net/isc-dhcp/files/dhcpd.init

```diff
@@ -1,33 +1,124 @@
 #!/bin/sh /etc/rc.common
 
-START=65
+START=19
+USE_PROCD=1
+PROG=/usr/sbin/dhcpd
 
 lease_file=/tmp/dhcpd.leases
-config_file=/etc/dhcpd.conf
-pid_file=/var/run/dhcpd.pid
+config_file=/tmp/run/dhcpd.conf
 
-start() {
-	if [ ! -e $lease_file ]; then
-		touch $lease_file
+num2ip() {
+	local network=$1
+	local num=$2
+
+	if [ $num -gt 65535 ] ; then
+		# class A network
+		echo "$(echo $network | cut -d'.' -f1).$(( num / 65536 )).$(( num / 256 % 256 )).$(( num % 256 ))"
+	elif [ $num -gt 255 ] ; then
+		# class B network
+		echo "$(echo $network | cut -d'.' -f1-2).$(( num / 256 % 256 )).$(( num % 256 ))"
+	else
+		# class C network
+		echo "$(echo $network | cut -d'.' -f1-3).$num"
 	fi
+}
+
+dhcpd_add() {
+	local cfg="$1"
+	local dhcp6range="::"
+	local ignore net networkid subnet ifname proto start limit end
+	local IP NETMASK BROADCAST NETWORK PREFIX DNS START END
+
+	config_get_bool ignore  "$cfg" 'ignore' 0
+	[ "$ignore" = "0" ] || return 0
+
+	config_get net "$cfg" interface
+	[ -n "$net" ] || return 0
+
+	config_get start "$cfg" start
+	[ -n "$start" ] || return 0
 
-	/usr/sbin/dhcpd -q -cf $config_file -lf $lease_file -pf $pid_file $dhcp_ifs
+	config_get limit "$cfg" limit
+	[ -n "$start" ] || return 0
 
-	if [ $? -ne 0 ]; then
-		return 1
+
+	network_get_subnet subnet "$net" || return 0
+	network_get_device ifname "$net" || return 0
+	network_get_protocol proto "$net" || return 0
+
+	[ static = "$proto" ] || return 0
+
+	dhcp_ifs="$dhcp_ifs $ifname"
+
+	eval "$(ipcalc.sh $subnet)"
+
+	if network_get_dnsserver dnsserver "$net" ; then
+		for dnsserv in $dnsserver ; do
+			[ -n "$DNS" ] && DNS="${DNS}, "
```

Comment:
>        ${parameter:+word}
>               Use  Alternate Value.  If parameter is null or unset, nothing is
>               substituted, otherwise the expansion of word is substituted.

In this case, either null or unset would be fine.